lib/pool.js

```diff
@@ -27,21 +27,19 @@ function _addBackoff(factory) {
 
   factory.create = function __run() {
 
-    return originalCreateMethod().catch((e) =>
+    return originalCreateMethod().catch(function ___handleErrorInCreate(e) {
 
-      new Bluebird(function ___backoffRunResolver(resolve, reject) {
-        const wait = backoff.next()
+      const wait = backoff.next()
```

Comment:
The back off is created with each call to the pool factory.  not each time the factory.create method is being run.  Therefore you get one back off instance per factory.create method.  Though now I see that I may need to add a `backoff.reset()` call on success.